stan/math/prim/scal/meta/partials_return_type.hpp

```diff
@@ -2,21 +2,50 @@
 #define STAN_MATH_PRIM_SCAL_META_PARTIALS_RETURN_TYPE_HPP
 
 #include <stan/math/prim/scal/meta/partials_type.hpp>
+#include <stan/math/rev/core/var.hpp>
+#include <stan/math/rev/scal/meta/partials_type.hpp>
+#include <stan/math/fwd/core/fvar.hpp>
+#include <stan/math/fwd/scal/meta/partials_type.hpp>
 #include <stan/math/prim/scal/meta/scalar_type.hpp>
 #include <boost/math/tools/promotion.hpp>
 
 namespace stan {
 
-template <typename T1, typename T2 = double, typename T3 = double,
-          typename T4 = double, typename T5 = double, typename T6 = double>
+/**
+ * Template metaprogram to calculate the base scalar return type resulting
```

Comment:
Just to clarify the doc a little, this computes the type of the partial derivative. Perhaps: "base scalar return type" -> "partial derivative type"

Maybe there's a better way to state that, but I think that would a little more accurate. I don't know if that'll only make sense to me. Please feel free to ignore or do replace with something else.

Reply:
Fixed